apps/accounts/views.py

```diff
@@ -71,7 +71,22 @@ def inner(self, request):
                                            config=settings.FXA_CONFIG)
         except verify.IdentificationError:
             return Response({'error': 'Profile not found.'}, status=401)
-        return fn(self, request, user=find_user(identity), identity=identity)
+        else:
+            identity_user = find_user(identity)
+            if request.user.is_authenticated():
```

Comment:
I stared at this if branch for a while because it feels like it should really be in `login_user()` but I realize it's also needed here. It just feels like `login_user()` depends on this check for safety. I guess it's ok to cement the inter-dependency with tests.